app/External/Stripe/SignatureValidator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\External\Stripe;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use Spatie\WebhookClient\SignatureValidator\SignatureValidator as SignatureValidatorBase;
+use Spatie\WebhookClient\WebhookConfig;
+
+//Set up using https://docs.stripe.com/webhooks?verify=verify-manually
+
+class SignatureValidator
+{
+    public function isValid(Request $request, WebhookConfig $config): bool
+    {
+        $signatureHeader = $request->header($config->signatureHeaderName);
+        $payload = $request->getContent();
+        $secret = $config->signingSecret;
+
+        $parsedResult = [];
+        parse_str(str_replace(',', '&', $signatureHeader), $parsedResult);
+
+        if (! array_key_exists('t', $parsedResult)) {
+            Log::info("Stripe webhook doesn't have timestamp");
+
+            return false;
+        }
+
+        if (! array_key_exists('v1', $parsedResult)) {
```

Comment:
I noticed the key you're checking for is `v1`, but the log comment suggests it's the signature and the parameter you're checking below is `signature`. Looking at the stripe docs, it suggests that `v1` is correct, but I'm wondering if we shouldn't use the built in libraries to validate this since it's more likely it'll be correct than if we do it all manually.